gc.c

```diff
@@ -4546,8 +4546,8 @@ cached_object_id(VALUE obj)
     else {
         GC_ASSERT(!FL_TEST(obj, FL_SEEN_OBJ_ID));
 
-        id = objspace->next_object_id;
-        objspace->next_object_id = rb_int_plus(id, INT2FIX(OBJ_ID_INCREMENT));
+        id = ULL2NUM(objspace->next_object_id);
+        objspace->next_object_id += OBJ_ID_INCREMENT;
```

Comment:
Needs overflow check?

Reply:
I don't think it's necessary. If we allocated 1000 object IDs every millisecond (i.e. 1,000,000 per second), it would take 15,000 years for it to overflow.

```
(((2**64) / 40) / ((1000 * 60 * 60 * 24) * 1000)) / 365
```